query/src/main/java/com/connexta/replication/query/QueryManager.java

```diff
@@ -0,0 +1,128 @@
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Managers for instantiating {@link QueryService} class based on configuration. */
+public class QueryManager {
+  private static final Logger LOGGER = LoggerFactory.getLogger(QueryManager.class);
+
+  /** Startup delay for polling for configuration changes in second. */
+  private static final long STARTUP_DELAY = TimeUnit.MINUTES.toSeconds(1);
+
+  private final QueryServiceTools queryServiceTools;
+
+  /** Sets of identifiers for the sites to be managed or empty to manage all sites. */
+  private final Set<String> sites;
+
+  private final ScheduledExecutorService executor;
+
+  /** Sets of active query services keyed by site its. */
+  @VisibleForTesting final Map<String, QueryService> services = new ConcurrentHashMap<>();
+
+  public QueryManager(Stream<String> sites, QueryServiceTools queryServiceTools) {
+    this(sites, queryServiceTools, Executors.newSingleThreadScheduledExecutor());
+  }
+
+  /**
+   * Instantiate a new query manager.
+   *
+   * @param sites identifiers for the sites to be managed or empty to manage all sites
+   */
+  QueryManager(
+      Stream<String> sites,
+      QueryServiceTools queryServiceTools,
+      ScheduledExecutorService executor) {
+    this.sites = sites.collect(Collectors.toSet());
+    this.queryServiceTools = queryServiceTools;
+    this.executor = executor;
+  }
+
+  /**
+   * Initializes the query manager and kick start all query services to start polling sites that
+   * must be polled.
+   */
+  public void init() {
+    if (sites.isEmpty()) {
+      LOGGER.info("Managing queries for all configured sites.");
+    } else {
+      LOGGER.info("Managing queries for sites: {}.", sites);
+    }
+    executor.scheduleAtFixedRate(
+        this::reloadSiteConfigs,
+        QueryManager.STARTUP_DELAY,
+        queryServiceTools.getPeriod(),
+        TimeUnit.SECONDS);
+  }
+
+  /** Destroys the query manager and stop all currently running query services. */
+  public void destroy() {
+    LOGGER.info("Shutting down query managers");
+    executor.shutdownNow();
+    for (QueryService service : services.values()) {
+      service.stop();
+    }
+  }
+
+  void reloadSiteConfigs() {
+    final Map<String, Site> newSites =
+        queryServiceTools.allSites().collect(Collectors.toMap(Site::getId, Function.identity()));
+
+    // stop query services for sites that are no longer configured or have changed to a non-polled
+    // type
+    // update them if they are already running
+    updateQueryServices(newSites);
+
+    // start query services for new sites that must be polled
+    newSites.values().stream()
+        .filter(QueryManager::mustBePolled)
```

Comment:
❗️ We should also filter for only the sites that are in our `sites` list if that one is not empty. Either here or at the line above where we get all sites from config.